tests/operator_test.go

```diff
@@ -261,6 +261,10 @@ var _ = Describe("[Serial]Operator", func() {
 
 					expectedID := curKv.Status.ObservedDeploymentID
 					if id != expectedID {
+						// terminating pods from previous installations don't receive any traffic anymore and can safely be jumped over
+						if pod.DeletionTimestamp != nil {
+							continue
```

Comment:
for us to get here, this pod that's set for deletion still has containers with the containerStatus.Ready=true and the pod itself is phase=running.

couldn't this could be an issue for leader election if this pod is the leader?

Reply:
I could not really find out what was causing this. The only explanation I have is test order change due to making some tests parallel executable. The interesting part is that I only run these tests after serial tests are done (on all lanes, since they start roughly at the same time). My refactoring changes nothing in the operator test part and there should not be more load than before.